cpp/src/arrow/compute/kernels/hash_aggregate_test.cc

```diff
@@ -614,6 +616,11 @@ TEST(Grouper, RandomStringInt64DoubleInt32Keys) {
   }
 }
 
+TEST(Grouper, NullKeys) {
+  TestGrouper g({null()});
+  g.ExpectConsume("[[null], [null]]", "[0, 0]");
```

Comment:
Is it possible to add a test with several keys and where one of them has the null type?

Reply:
More test cases including this one are added.